fsspec/caching.py

```diff
@@ -370,6 +370,16 @@ def __len__(self):
         return len(self.cache)
 
 
+class AllBytes(object):
```

Comment:
This is copied from the http implementation, which seems unnecessary duplication. Indeed, if we can fetch ranges, then this is not required anymore at all

Reply:
see my comment below about "copy-past". 
Also, your colleague advises me to put this class in the caching file. If the http implementation is modified to used it from the catching module, it will remove the redundancy problem at least for this part of the code and it seems also more coherent. But as the pull request is not about the http implementation, I didn't wanted to modify it my-self here. 
Tell me what you prefer about this please. :)

Reply:
Agree that this is clearly a caching method and should be included in the caching module, although it is not particularly useful for most files, since it just keeps the whole thing in memory. Moving the def out of the http module now is fine - there are tests for that functionality that would fail if somehow moving the code went wrong.